### PR TITLE
[client-v2] Compression Support

### PR DESCRIPTION
**Pull request by chernser**
## Summary

Implements support for next compression scenarios:
- whole http response from a server is compressed with LZ4
    - enabled when server &  http compression is enabled 
- response body from a server is compressed with LZ4
- http request is compressed with LZ4 
- request body is compressed with LZ4  

Closes https://github.com/ClickHouse/clickhouse-java/issues/1717

### CODE REVIEW

File: client-v2/src/main/java/com/clickhouse/client/api/internal/ClickHouseLZ4InputStream.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    private ByteBuffer buffer;
+
+    private byte[] tmpBuffer = new byte[1];
+
+
+    public ClickHouseLZ4InputStream(InputStream in, LZ4FastDecompressor decompressor) {
+        super();
+        this.decompressor = decompressor;
+        this.in = in;
+        this.buffer = ByteBuffer.allocate(8192);
```
Comment (mzitnik) [inline client-v2/src/main/java/com/clickhouse/client/api/internal/ClickHouseLZ4InputStream.java:33]: Let's extract it to const & make configurable

**Review by mzitnik — APPROVED** [review]
In general LGTM, but we need to add tests that can mimic different chunk sizes to simulate edge cases for the compression.
